spec/public/versions_spec.rb

```diff
@@ -78,4 +78,37 @@
       end
     end
   end
+
+  describe '#in_range' do
+    let(:versions) do
+      [
+        ResourceRegistry::Versions::Version.new('2024-01-01'),
+        ResourceRegistry::Versions::Version.new('2024-04-28'),
+        ResourceRegistry::Versions::Version.new('2024-09-20'),
+        ResourceRegistry::Versions::Version.new('2025-01-09')
+      ]
+    end
+
+    it 'filters versions by >= from & <= to' do
+      expect(subject.in_range('2024-04-28', '2024-09-20').count).to eq(2)
+    end
+
+    context 'with only from' do
+      it 'filters versions by >= from' do
+        expect(subject.in_range('2024-04-28', nil).count).to eq(3)
+      end
+    end
+
+    context 'with only to' do
+      it 'filters versions by <= to' do
+        expect(subject.in_range(nil, '2024-09-20').count).to eq(3)
```

Comment:
what if `subject.in_range(nil, nil)`?

Reply:
or would you expect a different behavior?

Reply:
I think it would be great to have this case covered since is the only one this test is missing and is one of the possible combinations :)

Reply:
Done!